### ISSUE TITLE
Paper "Analyzing Inverse Problems with Invertible Neural Netowrks": Error in implementation

### ISSUE THREAD

**Issue by renatobellotti** [vislearn/analyzing_inverse_problems#4]
Hi,

I'm trying to implement the invertible network described in the paper in Tensorflow 2.
I am having some difficulties matching the descriptions of the loss functions with the code.

Especially, I think there might be an inconsistency in [this file](https://github.com/VLL-HD/FrEIA/blob/master/experiments/inverse_problems_science/train.py):

If I've understood correctly, the function `loss_reconstruction` (that is almost undescribed in the paper) seems to use the following layout for the values that are fed to the sampling process:

![Bildschirmfoto von 2019-12-06 12-24-45](https://user-images.githubusercontent.com/10884299/70319844-90c56900-1823-11ea-86c3-268fe18c8a6c.png)

However, the `train_epoch` function seems to use a different layout:

![Bildschirmfoto von 2019-12-06 12-25-05](https://user-images.githubusercontent.com/10884299/70319870-a470cf80-1823-11ea-973d-b3f288de951b.png)

Is this a mistake, or does the output of the forward process really have a different format than the input of the inverse process?

**Comment by krishnarld** [comment]
> Hi,
> 
> I'm trying to implement the invertible network described in the paper in Tensorflow 2. I am having some difficulties matching the descriptions of the loss functions with the code.
> 
> Especially, I think there might be an inconsistency in [this file](https://github.com/VLL-HD/FrEIA/blob/master/experiments/inverse_problems_science/train.py):
> 
> If I've understood correctly, the function `loss_reconstruction` (that is almost undescribed in the paper) seems to use the following layout for the values that are fed to the sampling process:
> 
> ![Bildschirmfoto von 2019-12-06 12-24-45](https://user-images.githubusercontent.com/10884299/70319844-90c56900-1823-11ea-86c3-268fe18c8a6c.png)
> 
> However, the `train_epoch` function seems to use a different layout:
> 
> ![Bildschirmfoto von 2019-12-06 12-25-05](https://user-images.githubusercontent.com/10884299/70319870-a470cf80-1823-11ea-973d-b3f288de951b.png)
> 
> Is this a mistake, or does the output of the forward process really have a different format than the input of the inverse process?

Hi,

I have the same confusion. Have you figured it out? Please let me know. I could really use your help. Thanks.
